cloud/services/loadbalancers.go

```diff
@@ -102,11 +109,125 @@ func CreateNodeBalancerConfig(ctx context.Context, clusterScope *scope.ClusterSc
 		Check:     linodego.CheckConnection,
 	}
 
-	if linodeNBConfig, err = clusterScope.LinodeClient.CreateNodeBalancerConfig(ctx, clusterScope.LinodeCluster.Spec.Network.NodeBalancerID, createConfig); err != nil {
+	if linodeNBConfig, err = clusterScope.LinodeClient.CreateNodeBalancerConfig(
+		ctx,
+		clusterScope.LinodeCluster.Spec.Network.NodeBalancerID,
+		createConfig,
+	); err != nil {
 		logger.Info("Failed to create Linode NodeBalancer config", "error", err.Error())
 
 		return nil, err
 	}
 
 	return linodeNBConfig, nil
 }
+
+// GetNodeBalancerConfig gets the first NodeBalancer config
+func GetNodeBalancerConfig(
+	ctx context.Context,
+	clusterScope *scope.ClusterScope,
+	logger logr.Logger,
+) (*linodego.NodeBalancerConfig, error) {
+	var linodeNBConfigs []linodego.NodeBalancerConfig
+	var err error
+
+	if linodeNBConfigs, err = clusterScope.LinodeClient.ListNodeBalancerConfigs(
+		ctx,
+		clusterScope.LinodeCluster.Spec.Network.NodeBalancerID,
+		linodego.NewListOptions(1, ""),
+	); err != nil {
+		logger.Info("Failed to list NodeBalancer Configs", "error", err.Error())
+
+		return nil, err
+	}
+	if len(linodeNBConfigs) == 0 {
+		return nil, err
+	}
+
+	return &linodeNBConfigs[0], err
+}
+
+// AddNodeToNB adds a backend Node on the Node Balancer configuration
+func AddNodeToNB(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	clusterScope *scope.ClusterScope,
+	linodeInstance *linodego.Instance,
+) error {
+	// Update the NB backend with the new instance if it's a control plane node
+	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
+		return nil
+	}
+	// Get the private IP that was assigned
+	ips := linodeInstance.IPv4
+	if len(ips) < expectedIPV4Count {
+		err := errors.New("no private IP address")
```

Comment:
Are we able to assert that there's no private IP address if we have fewer than 2 addresses? Is it also possible that the only address in the machine spec is a private IP?

Reply:
I don't think that should be possible to have an instance created without a public IP, but to be safe I can add another [API call](https://pkg.go.dev/github.com/linode/linodego#Client.GetInstanceIPAddresses) to actually return the IPs explicitly for public vs private

Reply:
I'm pretty sure you can do it if you provision a Linode inside a VPC. 

The *net.IP struct has an IsPrivate() method that you can use to determine public vs private. That should work for the VPC address ranges too.

Reply:
Ah gotcha. I switched to just getting the private address(es) for the instance instead of checking each IP in the range of assigned addresses  for the instance.